### ISSUE TITLE
Firefox 42+ does not allow unsigned addons

### ISSUE THREAD

**Issue by Kermithenson** [YePpHa/YouTubeCenter#1992]
See: https://wiki.mozilla.org/Addons/Extension_Signing

![Blocked unsigned addon](http://i.imgur.com/KJwfswQ.png)

With the release of FF42 to nightly, unsigned addons are no longer loaded and there is no config flag to override. Nightly/Developer builds should migrate to AMO like tabmixplus [did](https://addons.mozilla.org/en-us/firefox/addon/tab-mix-plus/versions/) through the beta release channel: https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Maintenance#How_do_I_submit_a_Beta_add-on.3F

**Comment by Yonezpt** [comment]
>and there is no config flag to override.

That is not true, read https://github.com/YePpHa/YouTubeCenter/issues/1676#issuecomment-127994726

The inability to run unsigned add-ons will only be applied to stable and beta versions of Firefox, all the others have the ability to disable that restriction.

**Comment by Rwantare** [comment]
So, effectively, to use the developer version, I must shift to the Nightly channel.

**Comment by Yonezpt** [comment]
@Rwantare To use the add-on version you will have to use any non stable and non beta versions of Firefox. If you use the userscript version you can use any Firefox version.

**Comment by Rwantare** [comment]
@Yonezpt Thanks for the reminder, I've been glossing over that bit all along. I assume there is no performance hit for running it through Greasemonkey

**Comment by bcphung** [comment]
For those who can not install the add-on, you guys might find this: `xpinstall.signatures.required` in `about:config`, set it to `false`.

**Comment by Pontiac76** [comment]
Untested (As I'm running FF39) but from what I read, that'll only last for
a while as they're going to be removing that option and forcing addons to
be signed.  (Seriously annoyed at that)

**Comment by gorbavet** [comment]
Firefox 41 stable to be released tomorrow will disable unsigned addons. Since the "stable" channel of YTC is so outdated, we pretty much have to stay on the dev version, which isn't signed right now. But could you maybe upload the dev versions to AMO? Presumably then they'd be signed and we can still get the latest dev versions. It actually looks like AMO doesn't currently even have the latest stable version.

**Comment by Yonezpt** [comment]
@gorbavet The enforced signing has been postponed: http://www.ghacks.net/2015/09/11/mozilla-postpones-firefox-add-on-signing-enforcement/

And from what I have seen, YTC would require a massive rewrite to be approved in AMO due to their overwhelming list of restrictions they have been imposing lately.

2.1.7 was already rejected because of one of these restrictions https://github.com/YePpHa/YouTubeCenter/issues/1992#issuecomment-142135697

I have no doubt that if YePpHa were to correct this issue now it would still be rejected by a pletora of new issues they consider to be justified.

**Comment by Erroneuz** [comment]
Firefox 43 has been released which requires xpinstall.signatures.required to be set to false in about:config for YTC dev to work. From Firefox 44, which is expected to be released late January 2016, will not have a workaround and all unsigned extensions will NOT work in Firefox.

**Comment by Piter432** [comment]
@Erroneuz: Unless you'll install an unbranded version of Firefox.

**Comment by gorbavet** [comment]
@Piter432 That's not exactly a realistic solution for most users. I don't think it helps anybody to frame it as an ultimatum between the browser itself and *one* addon.

**Comment by Yonezpt** [comment]
Currently the available options are:

 - Disable the forced singature setting in the `about:config` page (will only work until Firefox 44 stable)
 - Use the userscript version
 - Use a Firefox version that isn't Stable or Beta (you still have to disable the forced signature setting mentioned above)

Each one chooses the one he prefers.

**Comment by Erroneuz** [comment]
Which means using the userscript version is pretty much the only real choice. 
Any disadvantages to the userscript or can we maybe hope that YTC will leave dev build status and be signed?

**Comment by gorbavet** [comment]
@Erroneuz The userscript variant should function exactly the same. IIRC YTC started off as a userscript and then got packed into the various extensions. I've switched to it already. Given @YePpHa 's time constraints for this project, I believe it is the best option right now.

**Comment by Zarggg** [comment]
Unsigned extensions can still be used in Firefox 43 by setting the `xpinstall.signatures.required` preference to `false`. However, this preference may be removed in Firefox 44.

**Comment by Zarggg** [comment]
The only real disadvantages of using the Userscript version is the overhead in loading Greasemonkey/Tampermonkey and the fact that a duplicate instance of the script must be loaded into memory for each tab that contains a YouTube page or embedded video.
